components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -727,13 +734,30 @@ class GeckoEngineSession(
             notifyObservers { onFullScreenChange(fullScreen) }
         }
 
-        override fun onExternalResponse(session: GeckoSession, response: GeckoSession.WebResponseInfo) {
-            notifyObservers {
-                onExternalResource(
-                        url = response.uri,
-                        contentLength = response.contentLength,
-                        contentType = response.contentType,
-                        fileName = response.filename)
+        override fun onExternalResponse(session: GeckoSession, webResponse: WebResponse) {
+            with(webResponse) {
+                val contentType = extractMimeType(getHeaderValue(CONTENT_TYPE))
+                val contentLength = getHeaderValue(CONTENT_LENGTH)?.toLong()
+                val contentDisposition = getHeaderValue(CONTENT_DISPOSITION)
+                val url = webResponse.uri
+
+                val fileName = DownloadUtils.guessFileName(
+                    contentDisposition,
+                    destinationDirectory = null,
+                    url = url,
+                    mimeType = contentType
+                )
```

Comment:
Oh, too bad we do not get those things from GV anymore. Now we have to fish them out of the response ourselves.

Reply:
Yeah, there are a lot edge cases that we are going to encounter that were handle by GV. At least, we are finding bugs in our code https://github.com/mozilla-mobile/android-components/issues/8425